src/app/setup/aws-setup/aws-setup.component.ts

```diff
@@ -84,11 +90,35 @@ export class AwsSetupComponent implements OnInit {
   }
 
   onSubmit() {
-    this.cloudProvider.isSelected = 2;
-    this.cloudProvider.credential.default_region = this.form.value['region'];
-    this.cloudProvider.credential.access_key_id = this.form.value['accessKeyId'];
-    this.cloudProvider.credential.secret_access_key = this.form.value['secretAccessKey'];
-
-    this.cloudProviderChange.emit(this.cloudProvider);
+    // Validate AWS credentials before continuing
+    let credentials = this.cloudProvider.credential;
+    if (credentials.default_region && credentials.access_key_id && credentials.secret_access_key) {
+      AWS.config.update({
+        region: credentials.default_region,
+        accessKeyId: credentials.access_key_id,
+        secretAccessKey: credentials.secret_access_key
+      });
+
+      // Validate AWS credentials trying to get the list of instances from the AWS EC2 service
+      // Check whether there exists a specific API method to validate credentials
+      let ec2 = new AWS.EC2();
+      ec2.describeInstances({}, (err: AWSError /*, data */) => {
+        if (err) {
+          if (err.code === 'AuthFailure') {
+            this.formErrors['accessKeyId'] = err.message;
+          } else {
+            console.error(err.code, err.message, err);
+            this.formErrors['accessKeyId'] = "Unable to validate access credentials";
```

Comment:
The message "Unable to validate access credentials" could be interpreted as something that was the user's fault. I propose something like: "We couldn't contact AWS to validate your credentials"